### PR TITLE
chore: Version bump opendal 0.41.0 -> 0.43.0

### PR DESCRIPTION
**Pull request by vincentdephily**
Remove the `opendal/services-redis-rustls` feature as it's no longer available/needed.

Obsoletes #1957

### CODE REVIEW

**Comment by sylvestre** [comment]
A bunch of tasks are failing here

**Comment by vincentdephily** [comment]
I managed to reproduce the errors by building with rust 1.71, and they go away with 1.72 and higher.

What puzzles me is that the CI seems to be using 1.67.1, but I get a bunch of different errors with 1.67-1.72, including fairly basic things like the clap crates crates in our Cargo.toml having an MSRV of 1.70. Does the CI really manages to do some builds with 1.67 ? Is it resolving minimal dependencies maybe ?

What's sccache's MSRV update policy ? Note that working with anything older than 1.70 has become a pain, due to fetching crate metadata using git instead of sparse registry.

**Comment by vincentdephily** [comment]
Sorry, ignore my previous observation about failures with 1.67-1.70, I had locally updated all deps by mistake.

The build failure with my branch does get resolved with 1.72, but that's a more isolated problem.

**Comment by vincentdephily** [comment]
Fix proposed upstream in Opendal. For this MR we'll probably want to wait for the next Opendal release.

**Comment by sylvestre** [comment]
the maintainer of opendal is also a maintainer here :)
So, should not be hard ;)
cc @Xuanwo ;)

**Comment by Xuanwo** [comment]
> Fix proposed upstream in Opendal. For this MR we'll probably want to wait for the next Opendal release.

Let's coordinate this.

**Comment by Xuanwo** [comment]
OpenDAL v0.43.0 is currently up for voting and we anticipate its availability by next week.

**Comment by vincentdephily** [comment]
Rebased on `main`, and now upgrading to `0.43.0` instead, along with a few dependencies.

**Comment by vincentdephily** [comment]
Gah, I had done the necessary `ordered-multimap` downgrade locally but forgot to amend my commit, sorry. Should be fixed.

**Review by Xuanwo — APPROVED** [review]
LGTM!

**Pull request merged**
